### DISCUSSION TITLE
Can EBSDMeshGenerator be calculated using distributed mesh?

### DISCUSSION THREAD

**Discussion by PengWei97** [idaholab/moose#22135] [Q&A Meshing] [answered]
Hello,
Recently, I want to use EBSD data for the large-scale phase field simulation. However, I found that setting `parallel_type = distributed` in `EBSDMeshGenerator ` of [IN100-111grn.i](https://github.com/idaholab/moose/blob/next/modules/phase_field/examples/ebsd_reconstruction/IN100-111grn.i), and it shows the following error,
```bash
...
*** ERROR ***
/home/pw-moose/projects/luwu/grainGrowth/gNGs2022/ebsd_00/IN100-111grn.i:6: unused parameter 'Mesh/ebsd_mesh/parallel_type'
```
Then, the details of the output to th terminal are in [01.log](https://github.com/PengWei97/luwu/blob/gNGsGG2022/grainGrowth/gNGs2022/ebsd_00/01.txt)

Therefore, my question is whether it is possible to use distributed mesh in `EBSDMeshGenerator ` to effectively reduce memory usage, or whether only replicated mesh can be used in `EBSDMeshGenerator`.

Any suggestions or recommendations to fix these problems would be greatly appreciated.

Thanks

wei peng

**Comment by GiudGiud** [comment, marked as answer]
Parallel_type is a [Mesh] block parameter, not a [Mesh/some_generator] parameter

Looking at the code and the tests, distributed meshes are supported for this mesh generator. 
You can either pass this parameter at the right location or add `--distributed-mesh` to the command line

**Reply by PengWei97** [reply]
Great, thank you for your reply. I solved the problem yesterday by modifying the code of EBSDMeshGenerator to set up the distributed mesh. 

wei

**Reply by GiudGiud** [reply]
nice!!
